### PR TITLE
Add unspecified value handling

### PR DESCRIPTION
**Pull request by jarrettranderson**

### CODE REVIEW

File: ipv-core/src/main/java/com/ibm/whc/deid/providers/masking/IMEIMaskingProvider.java [hunk 1]
```diff
@@ -10,36 +10,53 @@
...
   }
 
   @Override
   public String mask(String identifier) {
-    String tac;
-    if (!IMEIMaskingProvider.identifier.isOfThisType(identifier)) {
+
+    if (identifier == null) {
+      debugFaultyInput("null");
       return null;
```
Comment (DenisRicard) [inline ipv-core/src/main/java/com/ibm/whc/deid/providers/masking/IMEIMaskingProvider.java:43]: I think we shouldn't return null here but instead follow what the "unspecifiedValueHandling" value says to do in this case.

File: ipv-core/src/main/java/com/ibm/whc/deid/providers/masking/SWIFTCodeMaskingProvider.java [hunk 2]
```diff
@@ -29,11 +30,31 @@
...
+    unspecifiedValueReturnMessage = configuration.getUnspecifiedValueReturnMessage();
   }
 
   @Override
   public String mask(String identifier) {
     initialize();
+
+    if (identifier == null) {
+      debugFaultyInput("null");
+      return null;
```
Comment (DenisRicard) [inline ipv-core/src/main/java/com/ibm/whc/deid/providers/masking/SWIFTCodeMaskingProvider.java:43]: Same comment as above

File: ipv-core/src/main/java/com/ibm/whc/deid/providers/masking/StatesUSMaskingProvider.java [hunk 3]
```diff
@@ -12,45 +12,59 @@
...
     }
   }
 
   @Override
   public String mask(String identifier) {
     initialize();
+
+    if (identifier == null) {
+      debugFaultyInput("null");
+      return null;
```
Comment (DenisRicard) [inline ipv-core/src/main/java/com/ibm/whc/deid/providers/masking/StatesUSMaskingProvider.java:50]: same as above

**Review by DenisRicard — CHANGES_REQUESTED** [review]
Added a comment around null input values and the "unspecifiedValueHandling" . We should settle this and make it apply everywhere and clarify in "https://test.cloud.ibm.com/docs/data-de-identification?topic=data-de-identification-masking-providers#handling-of-null-and-unrecognized-input-data-values"

no other comments. Rest is good.

File: ipv-core/src/main/java/com/ibm/whc/deid/providers/masking/IMEIMaskingProvider.java (hunk 1 above)
Comment (jarrettranderson) [reply ipv-core/src/main/java/com/ibm/whc/deid/providers/masking/IMEIMaskingProvider.java:43]: This is a safety check.  Currently, if the input value is null, no call is made to the providers.  The code that determines what should be masked doesn't consider this a leaf.  Most of the providers, but not all, currently have a check like this.

**Pull request closed without merging**